src/Components/test/testassets/BasicTestApp/CascadingValueTest/CascadingValueReceiveFixedByInterface.cshtml

```diff
@@ -11,7 +11,9 @@
 <p>
     @(nameof(CascadingValueReceiveFixedByInterface)) render count:
     <strong id="receive-by-interface-num-renders">@numRenders</strong>
-    <button id="decrement-count" onclick=@Ancestor.DecrementCount>Decrement</button>
+
+    @* Prevent rerendering this component when the button is clicked. *@
+    <button id="decrement-count" onclick=@(EventCallback.Factory.Create(Ancestor, Ancestor.DecrementCount))>Decrement</button>
```

Comment:
I'm guessing this is the best thing to do based on the original intent of the request. Our new and improved logic makes you use workarounds if you really want to prevent rendering.

Reply:
Yes, OK. I guess it's a bit of an odd test now it's covering a scenario that wouldn't normally exist.

It could be expressed more clearly by not trying to stop the rendering, but instead having another component that is also wired to the same cascading value, and observing that it re-renders when the cascading value changed.

Assuming you don't want to refactor the test (though please go ahead if you do want!), perhaps the commend could be tweaked to the following to keep a record of the intent:

```
@* Prevent the usual rerendering this component when the button is clicked, so that the test code can observe it re-renders because the cascading value changes. *@
```

Reply:
Hang on, I'm confused. I thought that `onclick=@Ancestor.DecrementCount` would be exactly equivalent to `onclick=@(EventCallback.Factory.Create(Ancestor, Ancestor.DecrementCount))`, since `Ancestor.DecrementCount.Target == Ancestor`, right? Both formats would result in `Ancestor` being the recipient of the event, wouldn't they?

I must be misunderstanding, but could you clarify why it's different?

Reply:
Because of the new compiler feature, this will generate:

```
EventCallback.Factory.Create(this, Ancestor.DecrementCount) 
```

In this case `this` will win out over the delegate target. If you expected different lets dicuss this in detail.

Reply:
Ah, I see. In our earlier iterations of this new event dispatching technique, we looked at the supplied `delegate.Target`, and if it was `IHandleEvent`, we let it win over `this`.

If our objective in general is "the event goes to whichever component declares the handler", then I would have expected this one to go to `Ancestor`. I'm trying to think of scenarios where you'd reference a methodgroup on another component instance and *not* expect that other component to receive the event, but can't think of any. Can you think of any cases where you'd want `this` to win over `delegate.Target` when `delegate.Target` is itself a component? If there are any such cases then the existing logic is probably right.

This is quite an edge case, so even if we wanted to change this, I think it would be totally OK to do it as a follow-up that goes in after preview 3.

Reply:
Ok, let's do your version of this for now. I think you're right and I'd rather ship in preview3 the best version of this.